src/test/scala/akka/stream/contrib/SortWithinSpec.scala

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.stream.contrib
+
+import akka.stream.scaladsl.{Keep, Source}
+import akka.stream.testkit.scaladsl.{TestSink, TestSource}
+import akka.stream.{Attributes, DelayOverflowStrategy}
+
+import scala.concurrent.duration._
+
+class SortWithinSpecAutoFusingOn extends {
+  val autoFusing = true
+} with SortWithinSpec
+
+class SortWithinSpecAutoFusingOff extends {
+  val autoFusing = false
+} with SortWithinSpec
+
+trait SortWithinSpec extends BaseStreamSpec {
+
+  "A stream via SortWithin" should {
+    "sort the elements published till given FiniteDuration" in {
+      val (source, sink) = TestSource.probe[Integer]
+        .via(SortWithin[Integer](500.millis))
```

Comment:
This is a difficult balance, we are running CI on Travis, and that is very unpredictable with timing. Expect that you can have a GC (or other pause) for 2 seconds at any time. Perhaps something in the middle of this and your original values?